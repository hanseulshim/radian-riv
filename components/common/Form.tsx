import * as React from 'react'
import Alert from 'components/common/Alert'

type Props = {
  id: string
  children: React.ReactNode
  onSubmit: () => Promise<void>
  alert: {
    type: string
    message: string
  }
}

export default function Form({ children, id, onSubmit, alert }: Props) {
  const handleForm = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    onSubmit()
  }
  return (
    <form id={id} onSubmit={handleForm}>
      {children}
      {alert && <Alert type={alert.type} message={alert.message} />}
    </form>
  )
}
