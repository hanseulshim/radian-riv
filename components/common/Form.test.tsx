import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Form from './Form'

describe('Form', () => {
  let component
  beforeAll(() => {
    const props = {
      id: 'test',
      children: <div>form</div>,
      onSubmit: async () => {},
      alert: {
        type: 'success',
        message: 'request successful'
      }
    }
    component = shallow(<Form {...props} />)
  })
  it('renders form with id', () => {
    expect(component.find('#test')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
