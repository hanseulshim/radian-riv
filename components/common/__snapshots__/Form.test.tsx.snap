// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<form
  id="test"
  onSubmit={[Function]}
>
  <div>
    form
  </div>
  <Alert
    message="request successful"
    type="success"
  />
</form>
`;
